ml-algorithms/src/main/java/org/opensearch/ml/engine/Predictable.java

```diff
@@ -31,7 +33,13 @@ public interface Predictable {
      * @param mlInput input data
      * @return predicted results
      */
-    MLOutput predict(MLInput mlInput);
+    default MLOutput predict(MLInput mlInput) {
+        throw new IllegalStateException("Method is not implemented");
```

Comment:
Added tests for this. No, this is a runtime error, compilation won't be impacted by this. Below are reasons: 
First of all, this is an interface method but not all subcalsses need to implement this method, e.g. RemoteModel. So we need to make it a default method, and this method can be accessed in all its implemented classes. Also this method needs return value, there are three options:
1. return null: this is not good since this might take long time for someone to identify the root cause.
2. return a manually created object: this is not good since we don't want to manually create an object.
3. throw exception: this is good since the exception and error message is clear and searchable for developer/user thus they can identify the issue quickly.

In fact, there's another option which is keep this method abstract and all subclasses needs to implement this and based on their case, they can throw UnsupportedOperationException which can also avoid accidentally abuse of the method, but this option we need to change quite a lot of files, and since this also works I choose this simpler option.